src/transformState.js

```diff
@@ -4,8 +4,35 @@
  * @param {Object} state
  * @param {Object[]} actions
  */
+
+const ACTION_ADD_PROPERTIES = 'addProperties';
+const ACTION_REMOVE_PROPERTIES = 'removeProperties';
+const ACTION_CLEAR = 'clear';
+
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    switch (action.type) {
+      case ACTION_ADD_PROPERTIES:
+        Object.assign(state, action.extraData);
+        break;
+
+      case ACTION_REMOVE_PROPERTIES:
+        action.keysToRemove.forEach((key) => {
+          delete state[key];
+        });
+        break;
+      case ACTION_CLEAR:
+        for (const key in state) {
+          delete state[key];
+        }
+        break;
+
+      default:
+        Error('unsupported value');
+    }
```

Comment:
i think this will only create an error but won't halt the program to show it, maybe use throw new error